hash_practice/exercises.py

```diff
@@ -14,6 +69,24 @@ def top_k_frequent_elements(nums, k):
         Space Complexity: ?
     """
     pass
+    # top_k_elements = []
+    # frequency_hash = {}
+    return_arr = []
+    nums_dict = {}
+    if len(nums) == 0:
+        return []
+    for num in nums:
+        if num not in nums_dict:
+            nums_dict[num] = 1
+        else:
+            nums_dict[num] +=1
+    for i in range (0,k):
+        highest = max(nums_dict, key=nums_dict.get)
+        
+        return_arr.append(highest)
+        nums_dict.pop(highest)
+
+    return return_arr
 
 
 def valid_sudoku(table):

```

Comment:
Looks like you're on the right track, but I suggest making a helper function for each of the 3x3 subgrids.